karyon3-core/src/main/java/com/netflix/karyon/governator/AutoBindModule.java

```diff
@@ -0,0 +1,197 @@
+package com.netflix.karyon.governator;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import com.google.common.collect.Lists;
+import com.google.common.reflect.ClassPath;
+import com.google.inject.AbstractModule;
+import com.google.inject.Module;
+import com.google.inject.util.Modules;
+import com.netflix.governator.annotations.AutoBindSingleton;
```

Comment:
This class in is Karyon-core and hence makes a dependency on governator-annotations's AutoBindSingleton class. If in the future AutoBindSingleton is removed from governator annotations (which is possible as it's not recommended for use), this will cause some incompatible governator/karyon release versions. How hard would it be to move this feature to a karyon-ext or karyon-legacy-support submodule?